tests/d3d12_pso.c

```diff
@@ -2703,3 +2703,442 @@ void test_topology_triangle_fan(void)
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_EndQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 3);
+
+    /* Ensure that slope-scaled depth bias and depth bias clamp also apply */
+    ID3D12GraphicsCommandList9_ClearDepthStencilView(command_list9, dsv_handle, D3D12_CLEAR_FLAG_DEPTH, 0.0f, 0, 0, NULL);
+
+    push_args.depth = 0.5f;
+    push_args.slope = 0.25f;
+    ID3D12GraphicsCommandList9_SetGraphicsRoot32BitConstants(command_list9, 0, sizeof(push_args) / sizeof(uint32_t), &push_args, 0);
+
+    ID3D12GraphicsCommandList9_RSSetDepthBias(command_list9, 0.0f, 0.0f, 0.0f);
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_RSSetDepthBias(command_list9, 0.0f, 0.0001f, 0.01f);
+
+    ID3D12GraphicsCommandList9_BeginQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 4);
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_EndQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 4);
+
+    /* Reset clamp, this should still pass */
+    ID3D12GraphicsCommandList9_RSSetDepthBias(command_list9, 0.0f, 0.0f, 0.01f);
+
+    ID3D12GraphicsCommandList9_BeginQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 5);
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_EndQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 5);
+
+    /* Reset everything to zero, this should now fail */
+    ID3D12GraphicsCommandList9_RSSetDepthBias(command_list9, 0.0f, 0.0f, 0.0f);
+
+    ID3D12GraphicsCommandList9_BeginQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 6);
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_EndQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 6);
+
+    /* Ensure that binding a pipeline without the dynamic depth bias flag ignores
+     * any previously set dynamic depth bias. This is broken on AMD native. */
+    ID3D12GraphicsCommandList9_ClearDepthStencilView(command_list9, dsv_handle, D3D12_CLEAR_FLAG_DEPTH, 0.5f, 0, 0, NULL);
+    ID3D12GraphicsCommandList9_RSSetDepthBias(command_list9, 0.0f, 0.0f, 0.0f);
+    ID3D12GraphicsCommandList9_SetPipelineState(command_list9, pso_static_bias);
+
+    push_args.depth = 0.5f;
+    push_args.slope = 0.0f;
+    ID3D12GraphicsCommandList9_SetGraphicsRoot32BitConstants(command_list9, 0, sizeof(push_args) / sizeof(uint32_t), &push_args, 0);
+
+    ID3D12GraphicsCommandList9_BeginQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 7);
+    ID3D12GraphicsCommandList9_DrawInstanced(command_list9, 3, 1, 0, 0);
+    ID3D12GraphicsCommandList9_EndQuery(command_list9, query_heap, D3D12_QUERY_TYPE_BINARY_OCCLUSION, 7);
+
+#if 0
```

Comment:
Should this test be noped out if both drivers are robust against it?

Reply:
I only did it because it is technically wildly out of spec, but can re-enable.